### PR TITLE
USHIFT-947: Replace gcloud ssh/scp with raw ssh/scp in reboot verify

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

File: ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh [hunk 1]
```diff
@@ -5,32 +5,31 @@ set -xeuo pipefail
...
-
-# Copy pull secret to user home
-cp "${CLUSTER_PROFILE_DIR}"/pull-secret "${HOME}"/pull-secret
-
 gcloud auth activate-service-account --quiet --key-file "${CLUSTER_PROFILE_DIR}"/gce.json
 gcloud --quiet config set project "${GOOGLE_PROJECT_ID}"
 gcloud --quiet config set compute/zone "${GOOGLE_COMPUTE_ZONE}"
 gcloud --quiet config set compute/region "${GOOGLE_COMPUTE_REGION}"
 
+IP_ADDRESS="$(gcloud compute instances describe ${INSTANCE_PREFIX} --format='get(networkInterfaces[0].accessConfigs[0].natIP)')"
```
Comment (pmtk) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:21]: I included these changes in my e2e PR.
For this line to work, cloud instance must be already working so `gcloud compute instances start` must moved before this.

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-reboot

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-reboot

File: ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh [hunk 2]
```diff
@@ -27,39 +27,32 @@ Host ${INSTANCE_PREFIX}
...
+ssh "${INSTANCE_PREFIX}" "sudo cat /var/lib/microshift/resources/kubeadmin/${IP_ADDRESS}/kubeconfig" >/tmp/kubeconfig
 
-export KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig
-echo "waiting 180s to accomodate for slow kubelet actions with topolvm pvc after reboot"
-oc wait --for=condition=Ready --timeout=180s pod/test-pod
-EOF
-chmod +x "${HOME}"/wait_for_pod_ready.sh
-scp "${HOME}"/wait_for_pod_ready.sh "${INSTANCE_PREFIX}":~/wait_for_pod_ready.sh
+set +e
+oc wait --kubeconfig=/tmp/kubeconfig --for=condition=Ready --timeout=120s pod/test-pod
```
Comment (pmtk) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:51]: I think we need to wait until MicroShift is ready.
Immediately upon start of api-server everything is "ready" because it wasn't reconciled yet, you can look at `crictl ps` to see that no Pod is running

File: ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh [hunk 3]
```diff
@@ -27,39 +27,32 @@ Host ${INSTANCE_PREFIX}
...
+res=$?
+set -e
 
-if ! ssh "${INSTANCE_PREFIX}" 'sudo ~/wait_for_pod_ready.sh'; then
+if [ "${res}" -ne 0 ]; then
```
Comment (pmtk) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:55]: Why not put `oc wait...` as `if` condition and remove `set +/-e` and `res=$?`

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-reboot pull-ci-openshift-microshift-release-4.13-e2e-reboot-arm pull-ci-openshift-microshift-release-4.12-e2e-reboot

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-reboot pull-ci-openshift-microshift-release-4.13-e2e-reboot-arm pull-ci-openshift-microshift-release-4.12-e2e-reboot

File: ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh [hunk 4]
```diff
@@ -27,39 +27,27 @@ Host ${INSTANCE_PREFIX}
...
 
-timeout=1200 # 20 minute wait.  
+# Steps may not be used more than once in a test, so this block duplicates the behavior of wait-for-ssh for reboot tests.
+timeout=300 # 5 minute wait.
+>&2 echo "Polling ssh connectivity before proceeding.  Timeout=$timeout second"
 start=$(date +"%s")
-until ssh "${INSTANCE_PREFIX}" 'true'; do
-  if (( $(date +"%s") - start >= timeout )); then
-    echo "timed out out waiting for ssh connection" >&2
+until ssh "${INSTANCE_PREFIX}" 'sudo systemctl start microshift';
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:37]: Starting microshift remotely? Can you explain what we're trying to do here?

Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:37]: Do we need a loop here? If MicroShift cannot be started once, why would it be started on the second attempt?

File: ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh [hunk 5]
```diff
@@ -27,39 +27,27 @@ Host ${INSTANCE_PREFIX}
...
-set -x
-systemctl start microshift
-
-export KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig
-echo "waiting 180s to accomodate for slow kubelet actions with topolvm pvc after reboot"
-oc wait --for=condition=Ready --timeout=180s pod/test-pod
-EOF
-chmod +x "${HOME}"/wait_for_pod_ready.sh
-scp "${HOME}"/wait_for_pod_ready.sh "${INSTANCE_PREFIX}":~/wait_for_pod_ready.sh
+ssh "${INSTANCE_PREFIX}" "sudo cat /var/lib/microshift/resources/kubeadmin/${IP_ADDRESS}/kubeconfig" >/tmp/kubeconfig
```
Comment (ggiguash) [inline ci-operator/step-registry/openshift/microshift/e2e/reboot-tests/verify/openshift-microshift-e2e-reboot-tests-verify-commands.sh:48]: I think there's a race condition here because MicroShift takes a few seconds after start to generate kubeconfig file, right?
